pinot-spi/src/main/java/org/apache/pinot/spi/config/table/SegmentZKMetadataConfig.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.table;
+
+// TODO Possibly redundant class. Use SegmentZKMetadata throughout?
+/**
+ * ZK properties that are to be logged into segment's metadata.properties
+ */
+public class SegmentZKMetadataConfig {
```

Comment:
nit: this class is a bit confusing with the SegmentZkMetadata. How about something specific like RealtimeOffsetConfig or directly pass in startOffset endOffset for now?

Reply:
I was going with the general idea that more ZK props will need to be logged hence the name. But will think about this and update

Reply:
Renamed to SegmentZKPropsConfig. Since these are segment properties. We can use this class to add more ZK props that are to be logged.